File: daily/2025/1920_Build_Array_from_Permutation.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    vector<int> buildArray(vector<int>& nums) {
        int n = nums.size();
        vector<int> ans(n);
        for (int i = 0; i < n; i++) ans[i] = nums[nums[i]];
        return ans;
    }
};

int main(){
    vector<int> nums = {0,2,1,5,3,4};
    // vector<int> nums = {5,0,1,2,3,4};
    Solution S;
    vector<int> ans = S.buildArray(nums);
    for (int i : ans) cout << i << " ";
    cout << endl;
    return 0;
}

File: daily_python/1920_Build_Array_from_Permutation.py
import os
import sys
from typing import List

class Solution:
    def buildArray(self, nums: List[int]) -> List[int]:
        n = len(nums)
        ans = [nums[nums[i]] for i in range(n)]
        return ans
    
def readDataSet(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read().strip()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            nums = list(map(int, lines[0].split('=')[1].strip()[1:-2].split(',')))
            dataset.append(nums)
    return dataset

if __name__ == '__main__':
    if (len(sys.argv) == 1):
        filename = os.path.splitext(__file__)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readDataSet(filename)
    results = []
    solution = Solution()
    for nums in dataset:
        results.append(solution.buildArray(nums))
    for index, result in enumerate(results):
        print(f'Example {index + 1} : {result}')